### DISCUSSION TITLE
Is there a standard way to execute python code on PC side via Micropython?

### DISCUSSION THREAD

**Discussion by sandric** [micropython/micropython#11414] [Using MicroPython]
Hi, I wonder if there is some standardized way to call python code on PC side (via serial or CDC I suppose) from inside Micropython?
I understand I could write simple python script that listens to `/etc/tty` device for input string, and then `eval` it I suppose. But wanted to be sure there's no existing standard way to perform such two-way cross talking. In particular I'm talking of scenario when you accessing micropython on mcu with repl - so that mcu now knows there's a repl with python executable, and existing socket to write into it.
Thx.

**Comment by synapticonashaikh** [comment]
I'm not what you are trying to achieve here, but maybe you can look into `port/unix`. It's a cross-compiled micropython for the Linux environment which you can run on your Linux PC.

**Reply by Josverl** [reply]
There are a few jupyter implementations that allow mixing of cells of code to be run on an MCU and on a local/remote jupyter/ipython host.

But those all have the host in control. I'm not sure what your use case is, so those may not be a match

But generally speaking id call any pc waiting for python or any other repl/shell commands over the network or serial a security incident waiting to happen, unless there is solid authentication and authorization implemented.
